src/signals.ts

```diff
@@ -0,0 +1,23 @@
+export enum SignalKind {
```

Comment:
The term "signal" may be excessive in this project. You use 2 words to call a product of a source: "component" and "signal". They mean the same thing, so I recommend using only 1 of them to make the code easier to understand.

Reply:
FYI we use these terms in FingerprintJS OSS and Pro:

- "source" a.k.a. "entropy source" — a function that collects a piece of information about the browser
- "component" a.k.a. "entropy component" — a piece of information returned by an entropy source
- "signal" — any of the above

The ambiguity is resolved, so I'm closing the conversation.